### DISCUSSION TITLE
Can't include it in a Laravel project running on App Platform of Digital Ocean.

### DISCUSSION THREAD

**Discussion by AdelinGhanaem** [spatie/laravel-translation-loader#160] [Bugs]
When building a project with this dependency and then replacing the `Illuminate\Translation\TranslationServiceProvider::class,` with 'Spatie\TranslationLoader\TranslationServiceProvider::class,' the application deployment crashes saying it cannot get a Database connection. And it is failing during the` php artisan package:discovery --anis `command. it seems that package discovery requires a DB connection which is not provided during build time. Is it possible that `Spatie\TranslationLoader\TranslationServiceProvider::class` is trying to get a connection to db before env variables that contains DB params are available?

**Comment by Deividas-Prisify** [comment]
Yes. I have same issue. Deploying on Laravel Forge. Deployment script sequence

1. composer install and autoload;
2. autoload runs through web.php file;
3. package tries to load translation from Database, but in that time DB connections is incorrect and error occurs:
```BASH
SQLSTATE[HY000] [1045] Access denied for user 'forge'@'localhost' (using password: NO) (Connection: mysql, .....
```

### SOLUTION

- Extended packages LanguageLine model with own custom model, defined it in _[config/translation-loader.php](https://github.com/spatie/laravel-translation-loader/blob/aaf88739159ec415137738e4cc2b93b179fc1094/config/translation-loader.php#L17)_ file.

> This is the model used by the Db Translation loader. You can put any model here that [extends](https://github.com/spatie/laravel-translation-loader/pulls?tab=readme-ov-file#installation) Spatie\TranslationLoader\LanguageLine.

My custom model overrides `getTranslationsForGroup()` method with **try{}..catch{}**. If error in occurs, this method just returns empty array (as no translations in database, after this package continues to load translations from file).
```PHP
<?php

namespace App\Models;

use Illuminate\Database\QueryException;
use Illuminate\Support\Arr;
use Illuminate\Support\Facades\Cache;
use Spatie\TranslationLoader\LanguageLine;

class Translation extends LanguageLine
{
    protected $table = 'language_lines';

    /**
     * @return array<string, string>
     */
    public static function getTranslationsForGroup(string $locale, string $group): array
    {
        return Cache::rememberForever(static::getCacheKey($group, $locale), function () use ($group, $locale) {
            // Trying to load translations from Database, if no connection then return empty array.
            try {
                $dbResult = static::query()
                    ->where('group', $group)
                    ->get()
                    ->reduce(function ($lines, self $languageLine) use ($group, $locale) {
                        $translation = $languageLine->getTranslation($locale);

                        if ($translation !== null && $group === '*') {
                            // Make a flat array when returning json translations
                            $lines[$languageLine->key] = $translation;
                        } elseif ($translation !== null && $group !== '*') {
                            // Make a nested array when returning normal translations
                            Arr::set($lines, $languageLine->key, $translation);
                        }

                        return $lines;
                    });

                return $dbResult ?? [];
            } catch (QueryException $e) {
                //dump($e->getMessage());
                return [];
            }
        });
    }
}
```

In free time i will create PR. Or maybe someone will suggest better solution.
